crates/snapbox/src/data.rs

```diff
@@ -145,6 +135,43 @@ impl Data {
             DataInner::Text(data) => data.clone().into_bytes(),
         }
     }
+
+    pub fn coerce(self, format: DataFormat) -> Result<Self, crate::Error> {
```

Comment:
Errors when consuming `self` need to be careful.  For UTF-8 conversions, the standard library makes the original value available.  Here, we don't get any benefit from the error message, so I think one of the following would be appropriate:
- `coerce(...) -> Result<Self, Self>`
- `try_coerce(...) -> Self`

Reply:
So we should mimic `try_text` in the sense that failures to coerce still yield a `Data`?

Reply:
I think so